runtime/src/main/java/org/corfudb/protocols/logprotocol/SMREntry.java

```diff
@@ -68,6 +77,11 @@
     @Getter
     public transient boolean haveUpcallResult = false;
 
+    /** If smrEntry is NOP for hole */
+    public static boolean isNOP(SMREntry smrEntry) {
+        return smrEntry == NOP;
+    }
+
```

Comment:
Yea can get rid of the NOP notion by propagating the holes to the Stream layer but then filtering them out just before consumption in the `StreamViewSMRAdapter`